agent/src/main/java/org/jfrog/teamcity/agent/docker/ArtifactoryDockerBuildProcess.java

```diff
@@ -0,0 +1,122 @@
+        ArtifactoryDependenciesClientBuilder dependenciesClientBuilder =
+                BuildInfoUtils.getArtifactoryDependenciesClientBuilder(serverConfig, runnerParameters, logger);
+
+        if (isDockerCommandPull(commandType)) {
+            // Perform pull.
+            buildInfo = executeDockerPull(runnerParameters, imageName, host, buildInfoClientBuilder, dependenciesClientBuilder);;
+        } else {
+            // Perform push.
+            buildInfo = executeDockerPush(runnerParameters, imageName, host, buildInfoClientBuilder, dependenciesClientBuilder);
+        }
+
+        // Fail run if no build was returned.
+        if (buildInfo == null) {
+            buildInfoLog.error("Build operation returned empty build-info");
+            return BuildFinishedStatus.FINISHED_FAILED;
+        }
+
+        return BuildFinishedStatus.FINISHED_SUCCESS;
+    }
+
+    private Build executeDockerPull(Map<String, String> runnerParameters, String imageName, String host,
+                                    ArtifactoryBuildInfoClientBuilder buildInfoClientBuilder,
+                                    ArtifactoryDependenciesClientBuilder dependenciesClientBuilder) {
+        // Get pull parameters.
+        String sourceRepo = RepositoryHelper.getResolutionRepository(runnerParameters, valueResolver);
+        String userName = runnerParameters.get(RunnerParameterKeys.RESOLVER_USERNAME);
+        String password = runnerParameters.get(RunnerParameterKeys.RESOLVER_PASSWORD);
+
+        return new DockerPull(buildInfoClientBuilder, dependenciesClientBuilder, imageName, host, sourceRepo, userName, password,
+                buildInfoLog, environmentVariables).execute();
+    }
+
+    private Build executeDockerPush(Map<String, String> runnerParameters, String imageName, String host,
+                                    ArtifactoryBuildInfoClientBuilder buildInfoClientBuilder,
+                                    ArtifactoryDependenciesClientBuilder dependenciesClientBuilder) {
+        // Get push parameters.
+        String targetRepo = RepositoryHelper.getTargetRepository(runnerParameters, valueResolver);
+        String userName = runnerParameters.get(RunnerParameterKeys.DEPLOYER_USERNAME);
+        String password = runnerParameters.get(RunnerParameterKeys.DEPLOYER_PASSWORD);
+
+        return new DockerPush(buildInfoClientBuilder, dependenciesClientBuilder, imageName,
+                host, getCommonArtifactPropertiesMap(runnerParameters), targetRepo, userName, password,
+                buildInfoLog, environmentVariables).execute();
+    }
+
+    /**
+     * Get a map of the commonly used artifact properties to be set when deploying an artifact.
+     * Build name, build number, vcs url, vcs revision, timestamp.
+     * @return Map containing all properties.
+     */
+    private ArrayListMultimap<String, String> getCommonArtifactPropertiesMap(Map<String, String> runnerParameters) {
```

Comment:
This method should replace the old ```getMatrixParams()``` method located in the ```BaseBuildInfoExtractor``` class.

Reply:
Done.